### PR TITLE
Code style and docs references

### PR DESCRIPTION
**Pull request by jirikuncar**
depends on https://github.com/davebshow/aiogremlin/issues/9

### CODE REVIEW

**Comment by jirikuncar** [comment]
@davebshow I have enabled doctests and pycodestyle checks. In the future it would be nice to add pydocstyle and strict documentation build after the davebshow/aiogremlin#9 is resolved.

File: goblin/element.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# -*- coding: utf-8 -*-
```
Comment (davebshow) [inline goblin/element.py]: Why define the coding here? utf-8 is default in Python 3.

File: goblin/mapper.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def map_vertex_to_ogm(result, props, element, *, mapping=None):
 
 # temp hack
 def get_hashable_id(val):
-    #Use the value "as-is" by default.
+    """Use the value "as-is" by default."""
```
Comment (davebshow) [inline goblin/mapper.py]: I don't think that comment is meant to be a docstring.

File: setup.py [hunk 3]
```diff
@@ -48,6 +48,7 @@
     exec(fp.read(), g)
     version = g['__version__']
 
+print(packages)
```
Comment (davebshow) [inline setup.py]: This looks like leftover "debugging".

**Comment by davebshow** [comment]
That is fine with me, overall this looks pretty good. Could you please add documentation about how to build project/docs. Maybe a contributing section...

File: setup.py (hunk 3 above)
Comment (jirikuncar) [reply setup.py]: You are right 👍

**Comment by jirikuncar** [comment]
@davebshow I will have way bigger PR just for documentation later. For now I have fixed your other comments.

**Comment by davebshow** [comment]
I'm a bit slammed today. Will finish review first thing tomorrow.

File: conftest.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import asyncio
+
 import pytest
 from gremlin_python.process.traversal import Cardinality
+
 from goblin import Goblin, driver, element, properties
-from goblin.driver import (
-    Connection, DriverRemoteConnection, GraphSONMessageSerializer)
+from goblin.driver import Connection, DriverRemoteConnection, \
+    GraphSONMessageSerializer
```
Comment (davebshow) [inline conftest.py]: I don't understand this change. Is this PEP8?

File: conftest.py [hunk 5]
```diff
@@ -13,13 +15,18 @@ def pytest_generate_tests(metafunc):
 
 
 def pytest_addoption(parser):
-    parser.addoption('--provider', default='tinkergraph',
-                     choices=('tinkergraph', 'dse',))
```
Comment (davebshow) [inline conftest.py:17]: I don't see any style improvement here, again, what motivates this revision?

File: conftest.py [hunk 6]
```diff
@@ -156,20 +165,34 @@ def remote_connection(event_loop, gremlin_url):
 @pytest.fixture
 def connection_pool(gremlin_url, event_loop, provider):
     return driver.ConnectionPool(
-        gremlin_url, event_loop, None, '', '', 4, 1, 16,
```
Comment (davebshow) [inline conftest.py:159]: Again I don't understand how this improves style

File: docs/conf.py [hunk 7]
```diff
@@ -39,7 +39,7 @@
     'sphinx.ext.autodoc',
     'sphinx.ext.intersphinx',
     'sphinx.ext.viewcode',
-    'alabaster'
+    'alabaster',
 ]
```
Comment (davebshow) [inline docs/conf.py:43]: Why trailing comment?

File: goblin/abc.py [hunk 8]
```diff
@@ -52,8 +52,8 @@ def validate_vertex_prop(self, val, card, vertex_prop, data_type):
                 vp = vertex_prop(data_type, card=card)
                 vp.value = self.validate(v)
                 vertex_props.append(vp)
-            val = manager.ListVertexPropertyManager(
-                data_type, vertex_prop, card, vertex_props)
+            val = manager.ListVertexPropertyManager(data_type, vertex_prop,
```
Comment (davebshow) [inline goblin/abc.py:55]: This is fine, but idk if it is necessary.

File: goblin/app.py [hunk 9]
```diff
@@ -40,15 +44,22 @@ def __init__(self, cluster, *, provider=provider.TinkerGraph, get_hashable_id=No
         self._aliases = aliases
 
     @classmethod
-    async def open(cls, loop, *, provider=provider.TinkerGraph,
-                   get_hashable_id=None, aliases=None, **config):
+    async def open(cls,
```
Comment (davebshow) [inline goblin/app.py:47]: Again, really what is the difference other than personal preference.

**Review by davebshow — COMMENTED** [review]
I just added a few comments. Most of the stuff here is fine, although I don't understand the motivation behind some of the changes. What is the editorconfig for?

File: conftest.py (hunk 4 above)
Comment (jirikuncar) [reply conftest.py]: I just run a default `isort`. Do you have any preference on the style: https://github.com/timothycrosley/isort#multi-line-output-modes

File: docs/conf.py (hunk 7 above)
Comment (jirikuncar) [reply docs/conf.py:43]: so you get single line diff when you add new item to a list.

File: goblin/abc.py (hunk 8 above)
Comment (jirikuncar) [reply goblin/abc.py:55]: `yapf` default, we can play a bit with weights if necessary

File: goblin/app.py (hunk 9 above)
Comment (jirikuncar) [reply goblin/app.py:47]: it's not my preference, but automatic code formatting (see `yapf`). I would like to take the personal preference out of table

File: conftest.py (hunk 4 above)
Comment (jirikuncar) [reply conftest.py]: No, it's `isort` (see https://github.com/timothycrosley/isort#multi-line-output-modes)

File: conftest.py (hunk 5 above)
Comment (jirikuncar) [reply conftest.py:17]: `yapf` default - no personal opinion

**Comment by jirikuncar** [comment]
@davebshow in order to get good results when building documentation it would be nice to merge https://github.com/davebshow/aiogremlin/pull/10 first.

**Comment by jirikuncar** [comment]
Also please let me know if you have any preference for `isort` and `yapf` configuration. I try to keep both tools with almost default configuration not to force my personal preferences.

File: conftest.py (hunk 4 above)
Comment (davebshow) [reply conftest.py]: I think I prefer: 
4 - Hanging Grid

**Comment by davebshow** [comment]
Let's just change the isort preference to hanging grid, and after we merge #10 for aiogremlin I will merge this.

**Comment by jirikuncar** [comment]
@davebshow I have changed the `.editorconfig` and reformatted the imports.

**Comment by jirikuncar** [comment]
@davebshow did you have time to look into it?

**Comment by davebshow** [comment]
Hey @jirikuncar, didn't realize you had made these changes. Thanks!

**Pull request merged**